demo/src/app/app.component.ts

```diff
@@ -12,21 +15,29 @@ export class AppComponent implements OnInit {
 
   components = componentsList;
 
-  constructor(private _analytics: Analytics, router: Router) {
-    router.events.subscribe(event => {
-      if (event instanceof NavigationEnd) {
-        const {fragment} = router.parseUrl(router.url);
+  constructor(
+    private _analytics: Analytics,
+    router: Router,
+    @Inject(DOCUMENT) document: any
+  ) {
+    router.events
+      .pipe(filter(event => event instanceof NavigationEnd))
+      .subscribe(event => {
+        const { fragment } = router.parseUrl(router.url);
         if (fragment) {
-          const element = document.querySelector(`#${fragment}`);
-          if (element) {
-            element.scrollIntoView();
-          }
+          setTimeout(() => {
```

Comment:
this is going to be fixed separately with router changes from `6.1` later, right ?

Reply:
Yes